src/exchanges.js

```diff
@@ -267,6 +274,12 @@ Publisher.prototype.close = async function() {
   if (this._connecting) {
     await this._connecting;
   }
+  if (this.__reconnectTimer) {
+    clearTimeout(this.__reconnectTimer);
+  }
+  if (_sleepingTimeout) {
+    clearTimeout(_sleepingTimeout);
```

Comment:
Is this one necessary? If not please don't clear it... I think it's only present if thing is sleeping for 12s, you are effectively aborting that with a close...

Reply:
In fact there could be a second publisher that breaks, because you close the first publisher. This is not good.

Reply:
Without this the tests hang for 12 seconds before finishing.